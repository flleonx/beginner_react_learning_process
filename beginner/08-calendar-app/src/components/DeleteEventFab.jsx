import { useDispatch } from 'react-redux';

import { calendarEventStartDelete } from '../actions/calendarEvent';

export const DeleteEventFab = () => {

  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(calendarEventStartDelete());
  };

  return (
    <button
      className="btn btn-danger fab-danger"
      onClick={handleDelete}
    >
      <i className="fas fa-trash"></i>
      <span> Eliminar evento</span>
    </button>
  )
};
