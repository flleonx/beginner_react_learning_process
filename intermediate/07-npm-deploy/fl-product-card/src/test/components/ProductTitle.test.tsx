import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {

  test('It should show <ProductTitle /> properly with the custom title', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom Product"/>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('It should show the component with the product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

});
