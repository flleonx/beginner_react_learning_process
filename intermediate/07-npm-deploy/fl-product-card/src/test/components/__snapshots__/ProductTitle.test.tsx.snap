// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle It should show <ProductTitle /> properly with the custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`ProductTitle It should show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;
